utils/meta/open_data_network/open_data.py

```diff
@@ -0,0 +1,177 @@
+    "geographic locations and boundaries", "services and amenities", "buildings", "transparency", "water", "surveys",
+    "base maps"
+]
+FILTERED_NAMES = [
+    "permit", "inspection", "blood", "covid", "certificate", "fire and spec ops", "code violation", "weeds",
+    "calendar"
+]
+FILTERED_TAGS = [
+    "ems", "survey", "vaccination", "fire", "fire investigations", "employee", "311", "council", "animal", "animals",
+    "alarm", "tax", "alert", "dog", "chemical exposure", "directory"
+]
+
+
+def get_data(search_term):
+    """Gets Open Data Network data using the Socrata api.
+
+    Args:
+        search_term (str): Search term to query.
+
+    Returns:
+        list: List of results from the query.
+    """    
+    api_url = f"http://api.us.socrata.com/api/catalog/v1?categories=Public%20Safety&q={search_term}&limit=10000"
+    response = requests.get(api_url)
+
+    if response.status_code != 200:
+        print("Response returned status code {response.status_code}")
+        quit()
+
+    response_json = json.loads(response.text)
+
+    return response_json["results"]
+
+
+def filter_data(dataset):
+    """Filters out irrelevant data.
+
+    Args:
+        dataset (dict): Information about a dataset.
+
+    Returns:
+        bool: True if data is relevant, False otherwise.
+    """    
+    domain_category = ""
+    resource_name = dataset["resource"]["name"].lower()
+    domain_tags = [tag.lower() for tag in dataset["classification"]["domain_tags"]]
+
+    try:
+        domain_category = dataset["classification"]["domain_category"].lower()
+        # Check if the category needs to be filtered out
+        if any(tag in domain_category for tag in FILTERED_CATEGORIES):
+            # Transportation is a mostly irrelevant category, unless the name contains "crashes"
+            if domain_category == "transporation" and "crashes" in resource_name:
+                return True
+            else:
+                return False
+    except KeyError:
+        pass
+    
+    # Check if the name or tag needs to be filtered out
+    if any(name in resource_name for name in FILTERED_NAMES) or any(tag in domain_tags for tag in FILTERED_TAGS):
+        return False
+
+    return True
+
+
+def remove_duplicates(data):
+    """Removes data sources that are already in the PDAP database.
+
+    Args:
+        data (list): List of dataset dictionaries.
+
+    Returns:
+        list: List of datasets with duplicates removed.
+    """    
+    with open("PDAP Data Sources.csv", encoding="utf-8-sig") as data_sources:
+        sources_list = list(csv.DictReader(data_sources))
+
+    source_urls = list(map(lambda source: source["source_url"], sources_list))
+    data = filter(lambda dataset: dataset["link"] not in source_urls, data)
+
+    return list(data)
+
+
+def write_csv(data):
+    """Write the data to CSV file.
+
+    Args:
+        data (list): List of dataset dictionaries.
+    """    
+    fieldnames = [
```

Comment:
what if we left fieldnames we aren't importing out of this? smaller, more to-the-point file that's less brittle (schema changes)